### PR TITLE
[Gui] Make styles update on paint

### LINKED ISSUE #15756: [Regression] Stylesheet: Start page example files not legible with Dark Theme
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

The example file text from the Start page is white on white when using the FreeCAD Dark theme on macOS:
grafik

Using classic, the clicked example file gets a green border:
grafik

### Full version info

```shell
OS: macOS 13.6.7
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.36908 (Git)
Build type: Release
Branch: main
Hash: eb842aadffa2615037c45101ac2575fc79438850
Python 3.11.8, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * freecad.gears 1.2.0
  * OpenTheme 2024.7.30
  * CfdOF 1.27.4
  * Defeaturing 1.2.2
  * Manipulator 1.5.7
  * fasteners 0.5.25
  * CurvedShapes 1.0.9
  * sheetmetal 0.4.23
  * Curves 0.6.42
```

### Subproject(s) affected?

Core

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by benj5378**
Fixes #15756. Similar implementation to that already in FileCardDelegate.cpp.

### CODE REVIEW

**Comment by benj5378** [comment]
> The method fileCardStyle() returns an empty string if an application-wide style sheet is set. So, it should be fine to set the style sheet to these buttons if the returned string is not empty.

> This function is supposed to be used if no application-wide style sheet is used!

Alternative to #17347, for keeping the fallback function `fileCardStyles()`. I am really not in favor of this, as it makes preference duplication as explained in #14919. Why choose the more troublesome solution, with increased maintenance?

I don't see any reasoning for having `fileCardStyles()` and similar all over the codebase, instead of just requiring that a style sheet is always set.

On top of fixing #15756, this has a side effect: it makes the fallback styles actually work in Classic theme, as intially intended with the fallback styles. See the below screenshots.

### On first startup
**Before**
image
**After**
image

### On preceeding statups
**Before**
image
**After**
image

File: src/Mod/Start/Gui/StartView.cpp [hunk 1]
```diff
@@ -260,6 +248,17 @@ void StartView::configureNewFileButtons(QLayout* layout) const
     connect(arch, &QPushButton::clicked, this, &StartView::newArchFile);
 }
 
+void StartView::paintEvent(QPaintEvent* event)
```
Comment (benj5378) [inline src/Mod/Start/Gui/StartView.cpp:251]: If this seems silly, it is just the same fix as was already used in `FileCardDelegate.cpp`:

https://github.com/FreeCAD/FreeCAD/blob/ef2156d0f6d116b1642a5531a2d44fc6b7d17042/src/Mod/Start/Gui/FileCardDelegate.cpp#L85-L88

https://github.com/FreeCAD/FreeCAD/blob/ef2156d0f6d116b1642a5531a2d44fc6b7d17042/src/Mod/Start/Gui/FileCardDelegate.cpp#L115-L129

**Comment by yorikvanhavre** [comment]
@FreeCAD/design-working-group @wwmayer could you weight on these two solutions (this one and #17347 )? Which one do you prefer?

**Comment by MisterMakerNL** [comment]
I don't understand the before and after, like the cards are pushbuttons soo they should look like buttons and so in my opinion it should look like the second before:
image

**Comment by MisterMakerNL** [comment]
Cards like this have no signifiers that tell the user they can click on it, which is simply bad design.

**Comment by obelisk79** [comment]
Initial reaction is that the new looks worse than the old

**Comment by yorikvanhavre** [comment]
backported

**Comment by wwmayer** [comment]
This looks like a serious regression:
* Pre-selection of the top buttons works very unreliably
* It's extremely slow
*  One core of the processor is at 100% as long as the start page is open

**Comment by obelisk79** [comment]
This is interesting, because it's perfectly fine here with zero unusual performance or behavior.

**Comment by azuk** [comment]
I see the ~100% single CPU core usage as well and I bisected it to this change in the releases/FreeCAD-1-0 branch (cd99adc094a540370369ef0d8478c3048bc26415)

```
OS: Ubuntu 24.04.1 LTS (KDE/plasma/xcb)
Word size of FreeCAD: 64-bit
Version: 1.0.0RC2.38641 +362 (Git)
Build type: Debug
Branch: (no branch, bisect started on b0d847ca15)
Hash: cd99adc094a540370369ef0d8478c3048bc26415
Python 3.12.3, Qt 5.15.13, Coin 4.0.2, Vtk 9.1.0, OCC 7.8.1.dev
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
Installed mods: 
  * Render 2024.9.25
  * Curves 0.6.50
  * Assembly4 0.50.15
  * fasteners 0.5.29
  * Assembly3 0.12.2
```

**Comment by kadet1090** [comment]
With stylesheet it should not trigger so that is why not everyone sees the regression. Updating styles in paintEvent is a very risky move as paint events should be really fast and style recalculation is really slow. This should be reverted ASAP.

**Comment by benj5378** [comment]
I will have a look at this tomorrow. Wondering why it's not an issue here though
https://github.com/FreeCAD/FreeCAD/blob/ef2156d0f6d116b1642a5531a2d44fc6b7d17042/src/Mod/Start/Gui/FileCardDelegate.cpp#L85-L88

**Pull request merged**